From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <Spottedleaf@users.noreply.github.com>
Date: Sat, 19 Jun 2021 12:20:31 -0700
Subject: [PATCH] Fix BlockPos reobf problem

Non-reobf'd fields don't conflict, but reobf'd do! So
explicitly cast on field access to avoid.

diff --git a/src/main/java/net/minecraft/core/BlockPos.java b/src/main/java/net/minecraft/core/BlockPos.java
index b01d7da333bac7820e42b6f645634a15ef88ae4f..009febf8fccbfc9a638f018f921ebd9ec9068196 100644
--- a/src/main/java/net/minecraft/core/BlockPos.java
+++ b/src/main/java/net/minecraft/core/BlockPos.java
@@ -478,9 +478,9 @@ public class BlockPos extends Vec3i {
         }
 
         public BlockPos.MutableBlockPos set(int x, int y, int z) {
-            this.x = x; // Tuinity - force inline
-            this.y = y; // Tuinity - force inline
-            this.z = z; // Tuinity - force inline
+            ((Vec3i)this).x = x; // Tuinity - force inline
+            ((Vec3i)this).y = y; // Tuinity - force inline
+            ((Vec3i)this).z = z; // Tuinity - force inline
             return this;
         }
 
@@ -544,19 +544,19 @@ public class BlockPos extends Vec3i {
         // Paper start - comment out useless overrides @Override - TODO figure out why this is suddenly important to keep
         @Override
         public BlockPos.MutableBlockPos setX(int i) {
-            this.x = i; // Tuinity
+            ((Vec3i)this).x = i; // Tuinity
             return this;
         }
 
         @Override
         public BlockPos.MutableBlockPos setY(int i) {
-            this.y = i; // Tuinity
+            ((Vec3i)this).y = i; // Tuinity
             return this;
         }
 
         @Override
         public BlockPos.MutableBlockPos setZ(int i) {
-            this.z = i; // Tuinity
+            ((Vec3i)this).z = i; // Tuinity
             return this;
         }
         // Paper end
